### ISSUE TITLE
Encountered `Warning: Failed to converge after 100 iterations.`

### ISSUE THREAD

**Issue by kattni** [rvben/rumdl#336]
Hello!

I appear to have encountered a potential bug in rumdl, according to your  warning text. 

The rule codes listed are MD012 and MD013. 

It's continuing to happen if I run `fmt` again. I narrowed it down to the following error:

```
docs/en/news/posts/2023/buzz/february-2023-status-update.md:26:1: [MD013] Multi-line content can be normalized
```

It consisted of the last word of a sentence being on the line after the bulk of the sentence; I manually corrected the issue, and `fmt` ran successfully on the last few issues.

As I don't seem to have much other information to provide, I'm including a screenshot of the `rumdl check` immediately proceeding the `fmt`, and the `fmt` block where the error initially occurred.

<img width="1180" height="578" alt="Image" src="https://github.com/user-attachments/assets/9afb1a74-f5bf-491c-9e58-aac79b2dbaea" />

**Comment by martimlobao** [comment]
@kattni do you have an MWE you can share? an `example.md` and your config in `.rumdl.toml` would be helpful.

**Comment by kattni** [comment]
We shifted our rumdl config to `pyproject.toml`. I'm including that here. I had to rename it to `.txt` to get GitHub to play nice. It apparently doesn't like `.toml` files.

I manually fixed the issue yesterday, however, I remembered what the issue was, and was able to reproduce it. The `.md` file is included as well, with the issue in place.

[example.md](https://github.com/user-attachments/files/24849869/example.md)
[pyproject.txt](https://github.com/user-attachments/files/24849870/pyproject.txt)

**Comment by martimlobao** [comment]
Seems like the culprit is that trailing `2019.` you've got at the end of the list.

`mwe.md`
```markdown
# Title

2019.

## Subtitle

1. Lorem ipsum
```

`rumdl.toml`
```toml
[global]

[MD013]  # Line length
line_length = 999999
reflow = true
reflow_mode = "normalize"
```

```bash
❯ rumdl fmt
Warning: Failed to converge after 100 iterations.
This likely indicates a bug in rumdl.
Rule codes: MD012, MD013
Please report at: https://github.com/rvben/rumdl/issues/new
mwe.md:3:1: [MD013] Multi-line content can be normalized

Fixed: Fixed 1/1 issues in 1 file (35ms)
```

Still worth finding a reasonable fix here imo.

**Comment by rvben** [comment]
**This issue has been fixed in v0.1.1.**

**Root cause:** The text `2019.` at the end of a sentence was incorrectly identified as a numbered list item because `is_numbered_list_item()` accepted `digit.` without requiring a trailing space. This caused the reflow algorithm to misinterpret paragraph boundaries, leading to an oscillation between MD012 (blank lines) and MD013 (reflow) that never converged.

**Fix:** Commit 8930418c now requires a space after the period for numbered list detection, matching the CommonMark spec. `"2019."` is correctly treated as regular text, not a list item.

Thanks @martimlobao for the minimal reproduction case - that made it straightforward to identify and fix.

**Comment by kattni** [comment]
@rvben Thanks so much! @martimlobao Thank you for the assist!

@rvben I wanted to note that the link in the warning led me to a blank issue, and I forgot you had issue templates, so when I used the "Create more issues" option, I ended up filing four blank issues, instead of filing through your bug report template. If possible, you should consider updating that warning to point to your bug report template so you're getting more information from the start.

Cheers!

**Comment by rvben** [comment]
Thanks @kattni for that suggestion! That's something that I'd never have found myself. Recently I added templates and did not update that link. Updated and will be there next release. 👍
